### ISSUE TITLE
[Feature Request] Nested Browser

### ISSUE THREAD

**Issue by Yowlen** [SimpleMobileTools/Simple-Gallery#11]
First off, I love these tools. The K.I.S.S. philosophy (Keep It Simple, Stupid) is often forgotten by developers nowadays & the UI in these apps is wonderfully simple, which has a definite effect on ease-of-use. Keep up the good work.

Now, as for the actual request, I currently use QuickPic 4.5.2 (the last version before it was taken over by CM), mainly because it's the only gallery I've found that has a file manager-style nested browser. I would love to move to an open source one if only to stop Google's update annoyances with QuickPic.

**Comment by tibbi** [comment]
hey, glad you like it :) I was playing around with QuickPic, but Im not sure what do you mean. Could you please give some more details?

**Comment by Yowlen** [comment]
Basically, just a file manager. Having all the folders in one list like it currently is tends to clutter up the screen when there's a lot of folders with pictures in them. The nested view like a standard file manager is much more preferable in this case. As someone who doesn't have a desktop, I prefer to have all my pics easily accessible without internet, so a microSD card is my only option, and this layout helps a lot.

In QuickPic, the mode I'm talking about is enabled via the drop-down menu by going to Mode > Explorer.

**Comment by scalpremier** [comment]
Hi, I'm like Yowlen, I like this "Explorer Mode" in QuickPic.
I would like to know if you've planned to add this option in Simple Gallery ?
Thank you.

**Comment by tibbi** [comment]
Ye I plan adding more view modes, but maybe only in a couple months.

**Comment by TemporaryAccount0x1** [comment]
It's been almost 1 year any news on this ? 

This is a very useful feature since Simple Gallery would only need to load the contents of folders that a user presses on for example when a user is in the Picture folder only it's subfolders need to be shown.

**Comment by tibbi** [comment]
it isnt on the roadmap for the near future, it could still take a couple months.

**Comment by ceferrari** [comment]
That feature is all need to finally drop QuickPic and switch to Simple Gallery

**Comment by HoWilgh** [comment]
Hi tibbi,
I tested you Gallery app today and the first thing I miss is the feature described above.... The ability to show only the subfolders next to the pics of the current folder you are working in. Showing all folders and all pics on the phone in one long line makes it a bit cumbersome to navigate and find pics.
Thanks in advance

**Comment by tibbi** [comment]
you can enable direct subfolder grouping in Menu -> Change view type, in the pro version. That should help a bit

**Comment by HoWilgh** [comment]
Thank you for that comment.
I already had this setting enabled but it does not solve the problem. In my case is having pics sorted in subfolders turned into an disadvantage.

**Comment by Yowlen** [comment]
Yeah, I'm with HoWilgh. Having folders as a separate list doesn't give the same functionality or feel as having the folders as tappable items inside the list with the picture thumbnails and the back button serving as the "Go up a folder" function.

The goal here is to make it easy to go from folder to folder, and it's not as instinctive to navigate with the existing UI as it would be with the suggested UI.

**Comment by dharmendra1111** [comment]
This feature has a following problem :
--------------------------------------------------------------------------
Directory structure :
/storage/emulated/0/Pictures/Subfolder_1
/storage/emulated/0/Pictures/Subfolder_2
/storage/emulated/0/Pictures/Subfolder_3/Subfolder_3.1
/storage/emulated/0/Pictures/Subfolder_3/Subfolder_3.2

Resulting Display :
Pictures/(Subfolder_1 + Subfolder_2)
Subfolder_3/(Subfolder_3.1 + Subfolder_3.2)

Expected Display :
Pictures/(Subfolder_1 + Subfolder_2 + Subfolder_3/(Subfolder_3.1 + Subfolder_3.2))
--------------------------------------------------------------------------
Hopping for some update to correct this problem.

**Comment by tibbi** [comment]
the current solution says "Group DIRECT subfolders", so thats the way it is currently supposed to work. Subfolder_1 and Subfolder_3.1 are not direct subfolders.

**Comment by dharmendra1111** [comment]
> the current solution says "Group DIRECT subfolders", so thats the way it is currently supposed to work. Subfolder_1 and Subfolder_3.1 are not direct subfolders.

Thanks for reply.
I totally agree, and that is why I didn't say it's a bug. Instead it's a feature request, which most of users would want to have. Because current solution does the job partially by reducing some start up page folders.

**Comment by giesmininkas** [comment]
Hi! Thanks for the good work! I'm currently using SimpleGallery (was using QuickPic before) and also am missing this explorer-like interface option. While the "group direct subfolders" help somewhat, it's not a replacement for browser.
